compiler/src/dotty/tools/dotc/core/Designators.scala

```diff
@@ -23,8 +26,49 @@ object Designators {
 
     def asTerm(implicit ctx: Context): TermDesignator = unsupported("asTerm")
     def asType(implicit ctx: Context): TypeDesignator = unsupported("asType")
+
+    def withNameSpace(space: NameSpace)(implicit ctx: Context): Designator { type ThisName = self.ThisName } =
+      if (space == noNameSpace) this
+      else localName(this.asInstanceOf[ThisName], space)
+
+    /** Localize this name to the owner of `sym` if `sym` is private */
+    def localizeIfPrivate(sym: Symbol)(implicit ctx: Context): Designator { type ThisName = self.ThisName } =
+      if (sym.isPrivate) withNameSpace(sym.owner.typeRef) else this
+
+    def withSig(sig: Signature): Designator{ type ThisName = TermName } = {
+      val unsigned = this.asInstanceOf[TermName].exclude(SignedName)
+      if (sig eq Signature.NotAMethod) unsigned else SignedName(unsigned, sig)
+    }
   }
 
   type TermDesignator = Designator { type ThisName = TermName }
   type TypeDesignator = Designator { type ThisName = TypeName }
+
+  case class LocalName[N <: Name](name: N, nameSpace: TypeRef) extends Designator {
```

Comment:
Would be nice to document the concept of a LocalName.

Reply:
Also, it's a bit confusing that LocalName is not a subclass of Name, I would maybe call it LocalDesignator ?